### PR TITLE
Add and fix unit tests for AB#15564

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#15564](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15564)

## Description

- Fixed failing support service unit tests
- Fixed failing patient repository unit tests
- Added new unit tests for strategy pattern
- Added new unit tests for blocked access
- Added new unit tests for data sources
- Added new unit tests for audit repository
- Fixed improper spelling on SerializationHelperTests
- Fixed bug in support service code - use cache should be false
- Renamed method name in patient repository

**Unit Tests:**

Screenshot 2023-05-15 at 9 15 30 PM

Screenshot 2023-05-15 at 9 15 43 PM

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+            PatientRequest request = new(Hdid, true);
+
+            // Act
+            PatientModel? result = await hdidAllStrategy.GetPatientAsync(request).ConfigureAwait(true);
+
+            // Verify
+            Assert.Equal(Hdid, result?.Hdid);
+            Assert.Equal(Phn, result?.Phn);
+        }
```
Comment (BronzBierd) [inline Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs:100]: Could be tested as a `[Theory]`

File: Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+            PatientRequest request = new(Hdid, true);
+
+            // Act
+            PatientModel? result = await hdidPhsaStrategy.GetPatientAsync(request).ConfigureAwait(true);
+
+            // Verify
+            Assert.Equal(patientIdentity.HdId, result?.Hdid);
+            Assert.Equal(patientIdentity.Phn, result?.Phn);
+        }
```
Comment (BronzBierd) [inline Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs:114]: Could be tested as a `[Theory]`

**Review by BronzBierd — APPROVED** [review]
Only a few nitpicks, but they won't affect my approval, change them if you feel they're merited.

File: Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs [hunk 3]
```diff
@@ -219,6 +226,112 @@ async Task Actual()
             Assert.Equal(ErrorMessages.PhnInvalid, exception.ProblemDetails!.Detail);
         }
 
+        /// <summary>
+        /// Block access.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        [Fact]
+        public async Task ShouldBlockAccess()
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/PatientRepositoryTests.cs:234]: I think it should match the method it is testing.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs:100]: Same comment as previous.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs (hunk 2 above)
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs:114]: Same comment as previous.

File: Apps/AccountDataAccess/test/unit/Strategy/PhnEmpiStrategyTests.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+            PatientRequest request = new(Phn, true);
+
+            // Act
+            PatientModel? result = await phnEmpiStrategy.GetPatientAsync(request).ConfigureAwait(true);
+
+            // Verify
+            Assert.Equal(Hdid, result?.Hdid);
+            Assert.Equal(Phn, result?.Phn);
+        }
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/PhnEmpiStrategyTests.cs:88]: Same comment as previous.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidEmpiStrategyTests.cs [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+            PatientRequest request = new(Hdid, true);
+
+            // Act
+            PatientModel? result = await hdidEmpiStrategy.GetPatientAsync(request).ConfigureAwait(true);
+
+            // Verify
+            Assert.Equal(Hdid, result?.Hdid);
+            Assert.Equal(Phn, result?.Phn);
+        }
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/HdidEmpiStrategyTests.cs]: Changed.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/HdidAllStrategyTests.cs]: Changed.

File: Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs (hunk 2 above)
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/HdidPhsaStrategyTests.cs]: Changed.

File: Apps/AccountDataAccess/test/unit/Strategy/PhnEmpiStrategyTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/AccountDataAccess/test/unit/Strategy/PhnEmpiStrategyTests.cs]: Changed.

**Review by rayking29 — APPROVED** [review]

**Pull request merged**